libs/@local/repo-chores/scripts/prepublish.ts

```diff
@@ -0,0 +1,96 @@
+import path from "node:path";
+
+import chalk from "chalk";
+import execa from "execa";
+import fs from "fs-extra";
+
+import { UserFriendlyError } from "./shared/errors";
+import { checkIfDirHasUncommittedChanges } from "./shared/git";
+import {
+  derivePackageInfoFromEnv,
+  outputPackageInfo,
+} from "./shared/package-infos";
+import { updateJson } from "./shared/update-json";
+
+const script = async () => {
+  console.log(chalk.bold("Cleaning up before publishing..."));
+
+  const packageInfo = await derivePackageInfoFromEnv();
+  outputPackageInfo(packageInfo);
+
+  if (await checkIfDirHasUncommittedChanges(packageInfo.path)) {
+    throw new UserFriendlyError(
+      `Please commit or revert changes in ${packageInfo.path} before running this script`,
+    );
+  }
+
+  if (!(await fs.pathExists(path.resolve(packageInfo.path, ".npmignore")))) {
+    throw new UserFriendlyError(
+      `Please create .npmignore in ${packageInfo.path} before running this script`,
+    );
+  }
+
+  process.stdout.write(`Removing dist...`);
+
+  await fs.remove(path.resolve(packageInfo.path, "dist"));
+
+  process.stdout.write(" Done\n");
+  process.stdout.write(`Building...`);
+
+  // tsconfig.json is supposed to configured for local development and linting.
+  // We need to override some options to generate a build that is ready for publishing.
+  await execa(
+    "tsc",
+    // prettier-ignore
+    [
+      "--project", "tsconfig.json",
+      
+      "--declaration", "true",
+      "--jsx", "react-jsx",
```

Comment:
Do we need to overwrite this? Can we leave it up to the package to decide what it is? Presumably anything that is importing a published React component will itself be a React app and can bundle it, and preserving the JSX might make it slightly easier to debug / patch.

I know we had a conversation about this before but I can't find the PR it was on so I don't remember the details.

Reply:
Hmm I'm not sure if re can publish to `npm` with raw JSX in there. If we choose React though, the bundler will be able to replace it with Preact etc, if needed. Are you sure we published npm packages with JSX?

Reply:
We did at one point, although we only do it for the block templates now. I don't think there's anything preventing us from publishing things with preserved JSX if they're intended to be consumed as part of a React application anyway, but it's not important.